python/ray/tests/kuberay/test_kuberay_node_provider.py

```diff
@@ -291,6 +291,85 @@ def test_submit_scale_request(node_data_dict, scale_request, expected_patch_payl
         assert patch_payload == expected_patch_payload
 
 
+@pytest.mark.skipif(sys.platform.startswith("win"), reason="Not relevant on Windows.")
+@pytest.mark.parametrize(
+    "node_data_dict,scale_request,expected_patch_payload",
+    [
+        (
+            {
+                "raycluster-autoscaler-head-8zsc8": NodeData(
+                    kind="head",
+                    type="headgroup",
+                    replica_index=None,
+                    ip="10.4.2.6",
+                    status="up-to-date",
+                ),
+                "raycluster-autoscaler-worker-fake-gpu-group-2qnhv": NodeData(
+                    kind="worker",
+                    type="fake-gpu-group",
+                    replica_index=None,
+                    ip="10.4.0.6",
+                    status="up-to-date",
+                ),
+                "raycluster-autoscaler-worker-small-group-dkz2r": NodeData(
+                    kind="worker",
+                    type="small-group",
+                    replica_index=None,
+                    ip="10.4.1.8",
+                    status="up-to-date",
+                ),
+                "raycluster-autoscaler-worker-small-group-lbfm4": NodeData(
+                    kind="worker",
+                    type="small-group",
+                    replica_index=None,
+                    ip="10.4.0.5",
+                    status="up-to-date",
+                ),
+            },
+            ScaleRequest(
+                desired_num_workers={
+                    "small-group": 0,  # Delete 2
+                    "gpu-group": 1,  # Don't touch
+                    "blah-group": 5,  # Create 5
+                },
+                workers_to_delete={
+                    "raycluster-autoscaler-worker-small-group-lbfm4",
+                    "raycluster-autoscaler-worker-small-group-dkz2r",
+                },
+            ),
+            [
+                # The small-group is suspended. Therefore, no actual patch to it.
+                {
+                    "op": "replace",
+                    "path": "/spec/workerGroupSpecs/3/replicas",
+                    "value": 5,
+                }
```

Comment:
These test-generated patches do not include any patch related to the suspended small-group because we want to keep the original k8s CR values.